### PR TITLE
Modified python similarity:

### PR DESCRIPTION
**Pull request by Pilipets**
1. Added producing JSON to the output if called with --external;
2. Added capturing ValueError when computing TF-IDF for files without valid words;

Signed-off-by: Hlib Pylypets <pilipets.gleb@gmail.com>

### CODE REVIEW

File: web-service/api/python_wrappers/get_similarity_python.go [hunk 1]
```diff
@@ -14,6 +15,18 @@ const (
...
+}
+
+func NewPythonError(err_msg string) error {
+	return &PythonInternalError{err_msg}
+}
+
+func (e *PythonInternalError) Error() string {
+	return fmt.Sprintf("python scipt error: %v", e.ErrMsg)
+}
+
```
Comment (gevorgyana) [inline web-service/api/python_wrappers/get_similarity_python.go:29]: Can we just use objects with no pointers?

File: web-service/api/python_wrappers/get_similarity_python.go [hunk 2]
```diff
@@ -31,14 +44,14 @@ func GetPairwiseSimilarity(input_path string, args ...string) (string, error) {
 	}
 
 	args = append([]string{execute_path}, args...)
-	var pipe_out bytes.Buffer
+	var pipe_out, pipe_err bytes.Buffer
 	// rely on environment variable for `python`
 	cmd := exec.Command("python", append(args, input_path)...)
 	cmd.Stdout = &pipe_out
-	cmd.Stderr = os.Stderr
+	cmd.Stderr = &pipe_err
```
Comment (gevorgyana) [inline web-service/api/python_wrappers/get_similarity_python.go:51]: Using pipes here is an elegant solution.

File: web-service/api/python_wrappers/similarity_module.py [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 from sklearn.feature_extraction.text import TfidfVectorizer
 import matplotlib.pyplot as plt, numpy as np
 
+MAX_ALLOWED_FILES_SIZE = (1 << 20) * 10 # 10Mb
+
 def eprint(*args, **kwargs):
-    print('ERR %s: ' % datetime.now().strftime("%d/%m/%Y %H:%M:%S"), end='', file=sys.stderr)
     print(*args, file=sys.stderr, **kwargs)
-    sys.exit()
+    sys.exit(1)
```
Comment (gevorgyana) [inline web-service/api/python_wrappers/similarity_module.py:13]: I think it is the right way to end execution. Although exit() argument is a bit OS-dependent.

File: web-service/api/python_wrappers/similarity_module.py [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
 
 def cosine_similarity(file_objects, tokenizer_option):
-    remove_punct_dict = dict((ord(punct), None) for punct in string.punctuation)
+    remove_punct_dict = dict((ord(punct), None) for punct in string_punctuation)
```
Comment (gevorgyana) [inline web-service/api/python_wrappers/similarity_module.py:16]: :+1:

**Review by gevorgyana — APPROVED** [review]

File: web-service/api/python_wrappers/get_similarity_python.go (hunk 1 above)
Comment (Pilipets) [reply web-service/api/python_wrappers/get_similarity_python.go:29]: This is the interface method of the err, so I accept pointer here.

**Pull request merged**